### ISSUE TITLE
generated MO file is not deleted when its translation is removed

### ISSUE THREAD

**Issue by cfm** [WeblateOrg/weblate#8936]
### Describe the issue

With `weblate.gettext.mo` enabled, if a translation is added and subsequently removed, only its `.po` file, not its generated `.mo` file, is removed from the repository.

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

In a VCS-backed component from which Weblate pushes changes via pull request:

1. Add a new language `eo`.
2. Observe that `eo`'s `.po` and generated `.mo` files are pushed and present in the pull request.
3. Remove `eo`.

### Expected behavior

Both `eo`'s `.po` file and its generated `.mo` file are no longer present in the pull request.

### How do you run Weblate?

Docker container

### Weblate versions

Weblate 4.14.1; full `list_versions` output forthcoming.

**Comment by nijel** [comment]
Correct, the add-on currently does not support this. Removing files removed in Weblate would not be hard to implement, but that opens up additional questions. Should it also remove MO files for PO files removed in the repository? Should it enforce that MO files are consistent with the PO files?

**Comment by cfm** [comment]
That makes sense, @nijel.  Is the appropriate decluttering action then to remove the orphaned `.mo` file in the upstream repository?  (In this case, that would mean merging it first, since it's new in Weblate and doesn't yet exist upstream.)  If so, I'll bring a pull request to document that recommendation.  :-)

**Comment by nijel** [comment]
I think it makes sense that Weblate would remove the generated files when translation is removed. Currently, the add-ons do not have a hook for that, but it can be added. 

But still, there will be cases where manual cleanup will be needed, so documenting that is a good starting point and the documentation can be later adjusted once we implement a better solution.
